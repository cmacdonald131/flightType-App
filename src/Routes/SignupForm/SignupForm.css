* {
  box-sizing: border-box;
}

.wrapper {
  background: url(../../Images/wing.jpg);
  background-position: center center;
  background-size: cover;
  color: #ffffff;
  height: 100vh;
}

.SignupForm {
  text-align: center;
}

input,
label {
  display: block;
}

label {
  font-size: 18px;
  margin-bottom: 5px;
  margin-left: 10%;
}

button {
  padding: 5px 10px;
  margin-top: 15px;
}

input {
  padding-top: 5px;
  padding-bottom: 5px;
  margin-bottom: 10px;
  width: 80%;
  margin-left: auto;
  margin-right: auto;
  border: 1px solid black;
}

input::placeholder {
  padding-left: 5px;
  font-size: 16px;
}

@media screen and (min-width: 700px) {
  .wrapper {
    background: url(../../Images/flight.jpg);
    background-position: center center;
    background-size: cover;
    color: #ffffff;
    height: 100vh;
  }

  .formSection {
    width: 60%;
    margin-right: auto;
    margin-left: auto;
  }
}