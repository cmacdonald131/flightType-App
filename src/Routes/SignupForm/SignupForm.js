import React, { Component } from 'react'
import Navbar from '../../Components/Navbar/Navbar'
import AuthApiService from '../../Services/auth-api-service'
import './SignupForm.css'


export default class SignupForm extends Component {
    static defaultProps = {
        onRegistrationSuccess: () => { this.props.history.push('/login') }
    }

    state = { error: null }

    handleSubmit = ev => {
        ev.preventDefault()
        const { name, email, username, password, confirmPassword } = ev.target

        //confirms that the user input the correct password

        if (password.value !== confirmPassword.value) {
            console.log(password, confirmPassword)
            alert("Passwords don't match!")
        }

        //posts information to server as a new user

        this.setState({ error: null })
        AuthApiService.postUser({
            username: username.value,
            password: password.value,
            name: name.value,
            email: email.value,
        })
            .then(user => {
                name.value = ''
                email.value = ''
                username.value = ''
                password.value = ''
                confirmPassword.value = ''
                console.log(this.props.history.push('/login'))
                this.props.onRegistrationSuccess()
            })
            .catch(res => {
                this.setState({ error: res.error })
            })

    }

    render() {
        return (
            <div className="wrapper">
                <Navbar />
                <form
                    className='SignupForm'
                    onSubmit={this.handleSubmit}
                >
                    <header>
                        <h1>Sign up to get started!</h1>
                    </header>
                    <div className='formSection'>
                        <label htmlFor='SignupForm__uname' style={{ textAlign: "left" }}>
                            Name
                        </label>
                        <input
                            name='name'
                            type='text'
                            placeholder='Enter your Name'
                            required
                            id='SignupForm__name'
                        >
                        </input>
                    </div>
                    <div className='formSection'>
                        <label htmlFor='SignupForm__username' style={{ textAlign: "left" }}>
                            Username
                        </label>
                        <input
                            name='username'
                            type='text'
                            placeholder='Enter your Username'
                            required
                            id='SignupForm__username'
                        >
                        </input>
                    </div>
                    <div className='formSection'>
                        <label htmlFor='SignupForm__password' style={{ textAlign: "left" }}>
                            Password
                        </label>
                        <input
                            name='password'
                            type='password'
                            placeholder='Enter your Password'
                            required
                            id='SignupForm__password'
                            onClick={e => alert('Password must be at least 8 characters, contain no spaces, one upper case, lower case, number and special character.')}
                        >
                        </input>
                    </div>
                    <div className='formSection'>
                        <label htmlFor='SignupForm__confirmPassword' style={{ textAlign: "left" }}>
                            Confirm Password
                             </label>
                        <input
                            name='confirmPassword'
                            type='password'
                            placeholder='Confirm your Password'
                            required
                            id='SignupForm__confirmPassword'
                        >
                        </input>
                    </div>
                    <div className='formSection'>
                        <label htmlFor='SignupForm__email' style={{ textAlign: "left" }}>
                            Email
                            </label>
                        <input
                            name='email'
                            type='text'
                            placeholder='Enter your email'
                            required
                            id='SignupForm__email'
                        >
                        </input>
                    </div>
                    <button type='submit' className="btn" style={{ textAlign: "center" }}>
                        Register
                        </button>
                </form>
            </div>
        )
    }
}