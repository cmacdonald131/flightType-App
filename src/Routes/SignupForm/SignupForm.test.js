import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import SignupForm from './SignupForm'


describe(`SignupForm component`, () => {
  const props = {
    name: 'test-name',
    departure_airport: 'test-departure_airport',
    arrival_airport: 'test-arrival_airport',
    departure_time: 'test-departure_time',
    arrival_time: 'test-arrival_time',
    company: 'test-company',
    'data-other': 'test-other-prop'
  }

  it('renders a form.SignupForm by default', () => {
    const wrapper = shallow(<SignupForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the SignupForm given props', () => {
    const wrapper = shallow(<SignupForm {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})