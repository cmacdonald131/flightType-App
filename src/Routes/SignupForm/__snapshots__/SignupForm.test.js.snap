// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupForm component renders a form.SignupForm by default 1`] = `
<div
  className="wrapper"
>
  <Navbar />
  <form
    className="SignupForm"
    onSubmit={[Function]}
  >
    <header>
      <h1>
        Sign up to get started!
      </h1>
    </header>
    <div
      className="formSection"
    >
      <label
        htmlFor="SignupForm__uname"
        style={
          Object {
            "textAlign": "left",
          }
        }
      >
        Name
      </label>
      <input
        id="SignupForm__name"
        name="name"
        placeholder="Enter your Name"
        required={true}
        type="text"
      />
    </div>
    <div
      className="formSection"
    >
      <label
        htmlFor="SignupForm__username"
        style={
          Object {
            "textAlign": "left",
          }
        }
      >
        Username
      </label>
      <input
        id="SignupForm__username"
        name="username"
        placeholder="Enter your Username"
        required={true}
        type="text"
      />
    </div>
    <div
      className="formSection"
    >
      <label
        htmlFor="SignupForm__password"
        style={
          Object {
            "textAlign": "left",
          }
        }
      >
        Password
      </label>
      <input
        id="SignupForm__password"
        name="password"
        onClick={[Function]}
        placeholder="Enter your Password"
        required={true}
        type="password"
      />
    </div>
    <div
      className="formSection"
    >
      <label
        htmlFor="SignupForm__confirmPassword"
        style={
          Object {
            "textAlign": "left",
          }
        }
      >
        Confirm Password
      </label>
      <input
        id="SignupForm__confirmPassword"
        name="confirmPassword"
        placeholder="Confirm your Password"
        required={true}
        type="password"
      />
    </div>
    <div
      className="formSection"
    >
      <label
        htmlFor="SignupForm__email"
        style={
          Object {
            "textAlign": "left",
          }
        }
      >
        Email
      </label>
      <input
        id="SignupForm__email"
        name="email"
        placeholder="Enter your email"
        required={true}
        type="text"
      />
    </div>
    <button
      className="btn"
      style={
        Object {
          "textAlign": "center",
        }
      }
      type="submit"
    >
      Register
    </button>
  </form>
</div>
`;

exports[`SignupForm component renders the SignupForm given props 1`] = `
<div
  className="wrapper"
>
  <Navbar />
  <form
    className="SignupForm"
    onSubmit={[Function]}
  >
    <header>
      <h1>
        Sign up to get started!
      </h1>
    </header>
    <div
      className="formSection"
    >
      <label
        htmlFor="SignupForm__uname"
        style={
          Object {
            "textAlign": "left",
          }
        }
      >
        Name
      </label>
      <input
        id="SignupForm__name"
        name="name"
        placeholder="Enter your Name"
        required={true}
        type="text"
      />
    </div>
    <div
      className="formSection"
    >
      <label
        htmlFor="SignupForm__username"
        style={
          Object {
            "textAlign": "left",
          }
        }
      >
        Username
      </label>
      <input
        id="SignupForm__username"
        name="username"
        placeholder="Enter your Username"
        required={true}
        type="text"
      />
    </div>
    <div
      className="formSection"
    >
      <label
        htmlFor="SignupForm__password"
        style={
          Object {
            "textAlign": "left",
          }
        }
      >
        Password
      </label>
      <input
        id="SignupForm__password"
        name="password"
        onClick={[Function]}
        placeholder="Enter your Password"
        required={true}
        type="password"
      />
    </div>
    <div
      className="formSection"
    >
      <label
        htmlFor="SignupForm__confirmPassword"
        style={
          Object {
            "textAlign": "left",
          }
        }
      >
        Confirm Password
      </label>
      <input
        id="SignupForm__confirmPassword"
        name="confirmPassword"
        placeholder="Confirm your Password"
        required={true}
        type="password"
      />
    </div>
    <div
      className="formSection"
    >
      <label
        htmlFor="SignupForm__email"
        style={
          Object {
            "textAlign": "left",
          }
        }
      >
        Email
      </label>
      <input
        id="SignupForm__email"
        name="email"
        placeholder="Enter your email"
        required={true}
        type="text"
      />
    </div>
    <button
      className="btn"
      style={
        Object {
          "textAlign": "center",
        }
      }
      type="submit"
    >
      Register
    </button>
  </form>
</div>
`;
